orderbook/src/database/app_data.rs

```diff
@@ -0,0 +1,225 @@
+use super::*;
+use anyhow::Result;
+use futures::stream::TryStreamExt;
+use futures::StreamExt;
+use model::app_data::AppDataBlob;
+use primitive_types::{H160, H256};
+use std::borrow::Cow;
+use thiserror::Error;
+
+#[async_trait::async_trait]
+pub trait AppDataStoring: Send + Sync {
+    async fn insert_app_data(&self, app_data: &AppDataBlob) -> Result<H256, InsertionError>;
+    fn app_data<'a>(&'a self, filter: &'a AppDataFilter) -> BoxStream<'a, Result<AppDataBlob>>;
```

Comment:
Can't this also just be an async function that returns a Result? What is the value of returning a stream?

Reply:
The benefit I see is that it allows you to stream rows.

Otherwise, you could also just collect all rows into a `Vec` and make this an `async` function.

Reply:
Yeah, one could go with both ways. I used streaming, as I am not aware of any challenges with that and because it was used in most other instances as well. I thought there is value in having similar interfaces. But I can change, it is not important to me at all.

Reply:
I didn't know we were already using the BoxStream for orders and trades. In that case, let's keep it as is.